src/argilla/client/feedback/dataset/remote.py

```diff
@@ -0,0 +1,257 @@
+                client=self.client,
+                id=self.id,
+                offset=FETCHING_BATCH_SIZE * current_batch,
+                limit=FETCHING_BATCH_SIZE,
+            ).parsed
+            for record in batch.items:
+                yield self.__parse_record(record)
+            current_batch += 1
+
+            if len(batch.items) < FETCHING_BATCH_SIZE:
+                break
+
+    def add(
+        self,
+        records: List[FeedbackRecord],
+        show_progress: bool = True,
+    ) -> None:
+        for i in trange(
+            0, len(records), PUSHING_BATCH_SIZE, desc="Pushing records to Argilla...", disable=not show_progress
+        ):
+            records_batch = []
+            for record in records[i : i + PUSHING_BATCH_SIZE]:
+                if record.suggestions:
+                    for suggestion in record.suggestions:
+                        suggestion.question_id = self.__question_name2id[suggestion.question_name]
+                records_batch.append(
+                    record.dict(exclude={"id": ..., "suggestions": {"__all__": {"question_name"}}}, exclude_none=True)
+                )
+            datasets_api_v1.add_records(
+                client=self.client,
+                id=self.id,
+                records=records_batch,
+            )
+
+
+class RemoteFeedbackDataset(FeedbackDatasetBase):
+    def __init__(
+        self,
+        *,
+        client: "httpx.Client",
+        id: "UUID",
+        name: str,
+        workspace: "Workspace",
+        fields: List["AllowedFieldTypes"],
+        questions: List["AllowedQuestionTypes"],
+        guidelines: Optional[str] = None,
+    ) -> None:
+        super().__init__(fields=fields, questions=questions, guidelines=guidelines)
+
+        self.client = client
+        self._id = id
+        self._name = name
+        self._workspace = workspace
+
+        self._records: RemoteFeedbackRecords = RemoteFeedbackRecords(
+            client=self.client, id=self.id, questions=self.questions
+        )
+
+    @property
+    def records(self) -> RemoteFeedbackRecords:
+        return self._records
+
+    @property
+    def argilla_id(self) -> "UUID":
+        warnings.warn(
+            "`argilla_id` is deprected in favor of `id` and will be removed in a future"
+            " release. Please use `id` instead.",
+            DeprecationWarning,
+        )
+        return self.id
+
+    @property
+    def id(self) -> "UUID":
+        return self._id
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def workspace(self) -> "Workspace":
+        return self._workspace
+
+    @property
+    def url(self) -> str:
+        return f"{self.client.base_url}/dataset/{self.id}"
+
+    def __repr__(self) -> str:
+        return f"<ArgillaFeedbackDataset id={self.id} name={self.name} workspace={self.workspace}>"
+
+    def __len__(self) -> int:
+        return self._records.__len__()
```

Comment:
Is this the same as `len(self._records)`?

Reply:
Yes, but calling directly the magic method instead